### ISSUE TITLE
Sign macOS installers, Linux packages during official build on Arcade SDK

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#7018]
We may get this essentially for free in `master` and `release/3.0` because the Arcade SDK migration is able to use the Arcade signing infra. Filing this issue to track, but if it ends up not being trivial it'll be cut from the initial migration.

This issue doesn't track implementation on the servicing branches.

/cc @leecow @JohnTortugo

**Comment by JohnTortugo** [comment]
These two issues might provide some insight here: https://github.com/dotnet/arcade/issues/2852 https://github.com/dotnet/arcade/issues/2915

As far as I can think of, signing Linux packages (dep / rpm) is just a matter of specifying the right certificate in your eng\signing.props - Arcade doesn't have that configuration by default yet, I plan to do it soon. FYI: Lee currently uses this build definition to sign dep / rpm using Arcade SignTool: https://dnceng.visualstudio.com/internal/_build?definitionId=555&_a=summary

**Comment by dagood** [comment]
I had to revert macOS pkg installer signing because it broke the Core-Setup build in real-sign mode (https://github.com/dotnet/core-setup/pull/7306), and I had to revert Linux package signing because either signing silently did something bad or Core-SDK CI couldn't handle it (#7359).

We'll have to revisit this when there's more time to debug.

**Comment by dagood** [comment]
Closing, a tiny bit successful but the issue is also out of date:

On the Linux side, I was able to keep Deb/RPM package signing enabled in `master`, and Core-SDK took a fix to accept Msft-signed RPM packages and unblock dependency flow: https://github.com/dotnet/core-sdk/pull/3741#issuecomment-518797039. Deb/RPM signing is still disabled in `release/3.0`, and there's no pressing need to enable it.

On the macOS side, signing is be part of the larger effort to produce notarized macOS pkgs, which is tracked over here: https://github.com/dotnet/arcade/issues/3665.
